src/NServiceBus.Core/Pipeline/Incoming/IncomingContextImpl.cs

```diff
@@ -0,0 +1,85 @@
+namespace NServiceBus.Pipeline.Contexts
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Threading.Tasks;
+    using NServiceBus.Unicast;
+    using PublishOptions = NServiceBus.PublishOptions;
+    using ReplyOptions = NServiceBus.ReplyOptions;
+    using SendOptions = NServiceBus.SendOptions;
+
+    abstract class IncomingContextImpl : BehaviorContextImpl, IncomingContext
+    {
+        protected IncomingContextImpl(string messageId, string replyToAddress, IReadOnlyDictionary<string, string> headers, BehaviorContext parentContext)
+            : base(parentContext)
+        {
+
+            MessageId = messageId;
+            ReplyToAddress = replyToAddress;
+            MessageHeaders = headers;
+        }
+
+        /// <inheritdoc />
```

Comment:
`<inheritdoc />` actually doenst do anything

and no need for xml doco on an internal class